pkg/query/fanin.go

```diff
@@ -1,64 +1,132 @@
 package query
 
 import (
+	"sort"
 	"unsafe"
 
 	"strings"
 
 	"github.com/improbable-eng/thanos/pkg/store/storepb"
-	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/pkg/labels"
 	"github.com/prometheus/prometheus/storage"
-	"github.com/prometheus/tsdb"
 	"github.com/prometheus/tsdb/chunks"
-	"github.com/prometheus/tsdb/labels"
 )
 
-func mergeAllSeriesSets(all ...tsdb.SeriesSet) tsdb.SeriesSet {
+func mergeAllSeriesSets(all ...chunkSeriesSet) chunkSeriesSet {
 	switch len(all) {
 	case 0:
-		return errSeriesSet{err: nil}
+		panic("unexpected")
```

Comment:
why not errSet? (:

Reply:
can we not panic? ):

Reply:
Mh, so my understanding so far is that panic() is okay for severe contract violations. The standard library does it quite frequently then as well.

Changing it would require quite a bit of fiddling. I'll try to get that whole PR chain merged but will keep a mental note to possibly remove it within subsequent PRs.